src/unix/index_unix.ml

```diff
@@ -26,64 +26,41 @@ let current_version = "00000001"
 module Stats = Index.Stats
 
 module IO : Index.IO = struct
-  external ( ++ ) : int64 -> int64 -> int64 = "%int64_add"
-
-  external ( -- ) : int64 -> int64 -> int64 = "%int64_sub"
+  let ( ++ ) = Int64.add
 
   type t = {
     mutable file : string;
     mutable header : int64;
     mutable raw : Raw.t;
     mutable offset : int64;
-    mutable flushed : int64;
     mutable fan_size : int64;
     readonly : bool;
-    buf : Buffer.t;
     flush_callback : unit -> unit;
   }
 
-  let flush ?no_callback ?(with_fsync = false) t =
-    if t.readonly then raise RO_not_allowed;
-    let buf = Buffer.contents t.buf in
-    let offset = t.offset in
-    Buffer.clear t.buf;
-    if buf = "" then ()
-    else (
-      (match no_callback with Some () -> () | None -> t.flush_callback ());
-      Log.debug (fun l -> l "[%s] flushing %d bytes" t.file (String.length buf));
-      Raw.unsafe_write t.raw ~off:t.flushed buf;
-      Raw.Offset.set t.raw offset;
-      assert (t.flushed ++ Int64.of_int (String.length buf) = t.header ++ offset);
-      t.flushed <- offset ++ t.header);
-    if with_fsync then Raw.fsync t.raw
+  let flush ?(with_fsync = false) t = if with_fsync then Raw.fsync t.raw
 
   let rename ~src ~dst =
     flush ~with_fsync:true src;
     Raw.close dst.raw;
     Unix.rename src.file dst.file;
-    Buffer.clear dst.buf;
     src.file <- dst.file;
     dst.header <- src.header;
     dst.fan_size <- src.fan_size;
     dst.offset <- src.offset;
-    dst.flushed <- src.flushed;
     dst.raw <- src.raw
 
-  let close t =
-    if not t.readonly then Buffer.clear t.buf;
-    Raw.close t.raw
-
-  let auto_flush_limit = 1_000_000L
+  let close t = Raw.close t.raw
 
   let append t buf =
+    t.flush_callback ();
     if t.readonly then raise RO_not_allowed;
-    Buffer.add_string t.buf buf;
+    Raw.unsafe_write t.raw ~off:(t.header ++ t.offset) buf;
```

Comment:
This will cause a lot of short writes. Are you sure it's ok?

Reply:
It's not, I want to benchmark how much it is not :)

Reply:
sounds good :-)